Added Lottie animations, transition animations and card background

Saimun-Hasan opened the pull request:
This is my first open-source contribution. I liked the app and thought I should contribute in any way.
I tried to maintain existing code standards while updating the app. Feedback is welcome!

Changes Made:

- **Rapid Pass Background:** Added to the initial card for better aesthetics.
- **Text Overlay:** Implemented to improve text visibility.
- **Lottie Animations:** Integrated animations for NFC on/off states.
- **Transitions:** Added card sliding and balance expanding animations.

I ran the in built tests as well.
This was my first time doing App dev and kotlin. I watched couple videos and read documentations on what I wanted to do.

Screenrecording: 
https://github.com/user-attachments/assets/0feeb37c-52b4-41ec-9620-89392bc2d1b3

hmasum52 commented on app/src/main/res/drawable/rapid_pass_bg.png:
This a almost 814 KB png file. If you convert it to svg using any online tool and add it as vector image this will reduce the size. Also it will increase the resolution of the image as well.

I tired with your image and generated a 300kb svg!  

![image](https://github.com/user-attachments/assets/1163880a-8c18-4cc3-9b1c-3e564847f4ee)

> [!NOTE]
> I recommend using MRT pass image as background image though!

aniruddha-adhikary commented:
We don't have permissions to use the copyrighted design of Rapid Pass. it is an intellectual property of DBBL and associated companies. Let's use something generic!

hmasum52 commented:
@aniruddha-adhikary good catch! Haven't thought about the copy right issues :((

Saimun-Hasan commented:
@hmasum52  Thank you for the feedback.
@aniruddha-adhikary Thank you for pointing that out.

I took the inital image, removed the background just kept the metro pic. converted to xml and then used and pushed.
Current size is 59kb.

Please check if it works or not and or any updates.

aniruddha-adhikary commented:
This looks ✨ a m a z i n g ✨

But I am just worried this will make the flutter migration a bit tough. Could we rewrite this later in flutter?

Saimun-Hasan commented:
@aniruddha-adhikary I wish I could provide the answer.
this is me first time installing android studio and just getting kotlin patterns and reading doc to do 😓

After flutter conversion I can check, I didn't do much to begin with 😅
Should be able to do something

Maybe @hmasum52 can provide insight

hmasum52 commented:
There are flutter packages for lottie animation as well!

aniruddha-adhikary commented:
@Saimun-Hasan no need for a flutter rewrite but we are doing a major revision to the repo structure. Please hold on!

aniruddha-adhikary commented:
Sorry again, I will have to make you do a rebase / re open PR. Quite a bit has changed but the core is still kotlin!

Saimun-Hasan commented:
@aniruddha-adhikary  Not an issue!
Glad that I can still contribute even a little.
I'll close this PR 
I'll update re open on the new code base
